envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -89,6 +89,7 @@ class EnvoySnapshotFactory(
         }
 
         return addRemovedClusters(previousClusters, currentClusters)
+            .toSortedMap(Comparator.naturalOrder())
```

Comment:
This change may help remove superfluous changes but this code does not communicate intent. I see this `toSortedMap` and wonder why this needs to be a sorted map and why it's created in this place.

I'd also change return type of `clusterConfigurations` to `SortedMap<String, ClusterConfiguration>`.

Reply:
Added comment